### DISCUSSION TITLE
Can you provide a API to clear buffer immediately?

### DISCUSSION THREAD

**Discussion by carolin-violet** [sourcelair/xterm.js#5190] [Q&A]
since the websocket push so much data,when i manually close the websocket,the terminal still write data until all the websocket data is written to the terminal

**Comment by jerch** [comment]
> since the websocket push so much data,when i manually close the websocket,the terminal still write data until all the websocket data is written to the terminal

First - thats the intended behavior, as a wonky websocket shall not break the terminal. There are integrations, that reconnect quickly to keep a terminal working.
Second - your websocket should not be able to flood the terminal with data unconditionally, see [flow control](https://xtermjs.org/docs/guides/flowcontrol/).

> Can you provide a API to clear buffer immediately?

Maybe `terminal.reset` is what you are looking for (still 2. should be implemented).
